### PR TITLE
Add a PhoneNumber class to handle formatting and actions on numbers

### PR DESCRIPTION
**Pull request by ratamacue**
We value collaboration at Invoca, and focus on peer mentorship as a way for the whole team to learn and thrive. As part of this process, we conduct code reviews during our development cycle. In this phase of the interview, we would like you to review and comment on this Pull Request. You can assume that this Pull Request was written by an intern, who is new to professional software development.  Please take about 20-30 minutes to review this Pull Request and comment as if you were reviewing an intern's code.

### CODE REVIEW

File: phone_number.rb [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  def find_area_code
+    open_paren = @number.index('(')
+    close_paren = @number.index(')')
+    if open_paren
+      if close_paren
+        @number[(open_paren + 1)..(close_paren - 1)]
+      end
+    else
+      @number.gsub!(/[^0-9]/, "")
```
Comment (a613) [inline phone_number.rb:48]: My main concern with these methods is that they assume there are 10 digits in the string, but somebody may have put a shorter one in, or maybe they included the country code, and that will throw these specific indexes off. Can you ask about expected formats on the Jira, and we can have a call later to figure out how to handle strange inputs.

File: phone_number_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
```
Comment (a613) [inline phone_number_spec.rb:1]: have to `require 'rails_helper'` and `require_relative "./phone_actions"` too 👍 

But...actually could we split out the PhoneActions tests to another spec file to keep things clean?

File: phone_number_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+RSpec.describe PhoneNumber do
+  let(:phone_number) { PhoneNumber.new("5558675309") }
```
Comment (a613) [inline phone_number_spec.rb:4]: It's nice to have a single instance if we can do all the tests we need on it, but in this case it might be simpler to just make new instances in each test below

File: phone_number_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+require_relative "./phone_number"
+
+RSpec.describe PhoneNumber do
```
Comment (a613) [inline phone_number_spec.rb:3]: Can we add some more cases to these tests that use weird phone numbers like `1234`, `tel:+1 613-555-1212`, `+()-`, and even an empty string? This goes along with the question in the other comment about how bad the input might be.

File: phone_number_spec.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    it "is the number as an int" do
+      new_number = PhoneNumber.new("555-867-5309")
+      expect(new_number.to_i).to eq(5558675309)
+    end
+  end
+
+  describe "#text" do
+    it "calls PhoneActions#text!" do
+      expect_any_instance_of(PhoneActions).to receive(:text!)
+      phone_number.text("hello world!")
```
Comment (a613) [inline phone_number_spec.rb:22]: for these methods that do API calls, let's also

* test the response value
* test the response if an error happens

Comment (a613) [inline phone_number_spec.rb:22]: As written these are more of integration tests because it's going to try to actually send a text (or a call, for the other test) to whatever number is in the env file.

Let's mock out the API calls so we can see this working without actually doing network stuff. That will also make it easy to see what happens if the "API" returns an error response

File: phone_number.rb [hunk 6]
```diff
@@ -0,0 +1,63 @@
+require_relative "./phone_actions"
+
+class PhoneNumber
```
Comment (a613) [inline phone_number.rb:3]: would be nice to have a little docstring here and on PhoneActions

**Review by a613 — CHANGES_REQUESTED** [review]
Cool! Found a few things to sort out. This is a good first pass, thanks for getting it in early so I could have a read through before the weekend.

Let me know if there's anything that doesn't make sense, I didn't run this myself yet so I might have missed something.
